src/approaches/analysis_two_approach.py

```diff
@@ -0,0 +1,263 @@
+"""Open-loop large language model (LLM) meta-controller approach with prompt modification where operator names are randomly generated.
+Note: Predicate and Task classes must be set to frozen=False
+Example command line:
+    export OPENAI_API_KEY=<your API key>
+    python src/main.py --approach analysis_two --seed 0 \
+        --strips_learner oracle \
+        --env pddl_blocks_procedural_tasks \
+        --num_train_tasks 3 \
+        --num_test_tasks 1 \
+        --debug
+
+Easier setting:
+    python src/main.py --approach analysis_two --seed 0 \
+        --strips_learner oracle \
+        --env pddl_easy_delivery_procedural_tasks \
+        --pddl_easy_delivery_procedural_train_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_train_max_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_max_extra_newspapers 0 \
+        --num_train_tasks 5 \
+        --num_test_tasks 10 \
+        --debug
+"""
+from __future__ import annotations
+
+import logging
+from typing import Collection, Dict, List, Optional, Sequence, Set, Tuple
+
+from predicators.src.approaches import ApproachFailure
+from predicators.src.approaches.nsrt_metacontroller_approach import \
+    NSRTMetacontrollerApproach
+from predicators.src.llm_interface import OpenAILLM
+from predicators.src.planning import task_plan_with_option_plan_constraint
+from predicators.src.settings import CFG
+from predicators.src.structs import Box, Dataset, GroundAtom, Object, \
+    ParameterizedOption, Predicate, State, Task, Type, _GroundNSRT, _Option
+
+original_strings = []
+random_strings = []
+
+
+class AnalysisTwoApproach(NSRTMetacontrollerApproach):
```

Comment:
This seems to be sharing a lot of code with the `AnalysisOneApproach` (seems like almost all functions are the exact same except `_llm_prediction_to_option_plan` and `_create_prompt`?) and even the `LLMOpenLoopApproach`. I think things will be much cleaner if we share code amongst all of these in a nice way: e.g., maybe we should subclass `LLMOpenLoopApproach` into `LLMAnalysisApproach` and further subclass `AnalysisOneApproach` and `AnalysisTwoApproach` from this common superclass? Or maybe both these new analysis approaches should just directly subclass `LLMOpenLoopApproach`